src/tiled/tileseteditor.cpp

```diff
@@ -465,7 +465,28 @@ void TilesetEditor::addTiles()
         Tile *newTile = new Tile(tileset->takeNextTileId(), tileset);
         newTile->setImage(loadedFile.image);
         newTile->setImageSource(loadedFile.imageSource);
-        tiles.append(newTile);
+
+        // If the tile is already in the tileset, warn user and confirm addition
+        bool addImage = true;
+        const QMap<int, Tile*> addedTiles = mCurrentTilesetDocument->tileset().data()->tiles();
+        for(auto tile : addedTiles) {
+            if(tile->imageSource() == newTile->imageSource()) {
+                QMessageBox warning(QMessageBox::Warning,
+                            tr("Add Tiles"),
+                            tr("Tile already exists in the Tileset!"),
+                            QMessageBox::Yes | QMessageBox::No,
```

Comment:
Issue #1227 mentioned a "Cancel" action as well, and I think we should have it. The reason for this is that when the user selected a large amount of tiles and changes his mind because of duplicates, he needs to be able to abort the operation without clicking No a lot of times.

Ideally, the user should have the options:

* Don't add this one (No)
* Don't add any duplicates (No to All)
* Add this one (Yes)
* Add all (Yes to All, skipping duplication checks)
* Cancel

The "No to All" and "Yes to All" actions are sometimes implemented with some kind of "Don't ask again" checkbox, which could be added using `QMessageBox::setCheckBox`. I know Windows Explorer has something like this when you move multiple files and it finds existing ones, maybe we can copy the text it is using (can't remember it now).